packages/commerce/src/api/utils/with-schema-parser.ts

```diff
@@ -0,0 +1,54 @@
+import type { AllowedOperations, OperationsData } from '../operations'
+
+import { z } from 'zod'
+
+import { getOperationError } from './errors'
+import { pageSchema } from '../../schemas/page'
+import { siteInfoSchema } from '../../schemas/site'
+import { productSchema, productsPathsSchema } from '../../schemas/product'
+
+export const withSchemaParser =
+  (
+    operation: AllowedOperations,
+    fn: (...args: any[]) => Promise<OperationsData>
+  ) =>
```

Comment:
Could this method receive a type that represents the name of the operation and that has to extend `AllowedOperations`, and then the return type `Promise<OperationsData>` could be based on that name instead?

Reply:
Changed this to with-operation-callback.ts. In this method, we can chain and do other parallel async transformations to the output. For example, something I have done in another branch to add automatic blurDataURL to images.